internal/provider/resource_role.go

```diff
@@ -151,38 +151,38 @@ func resourceRoleCreate(ctx context.Context, d *schema.ResourceData, meta interf
 	if tcr == nil {
 		return diag.Errorf("nil role after create")
 	}
-	raw := tcr.GetResponse().Map
+	apiResponse := tcr.GetResponse().Map
+	defer func() {
+		if err := setFromRoleResponseMap(d, apiResponse); err != nil {
+			errs = append(errs, diag.FromErr(err)...)
+		}
+	}()
 
-	var diags diag.Diagnostics
 	if principalIds != nil {
 		tspr, err := rc.SetPrincipals(ctx, tcr.Item.Id, 0, principalIds, roles.WithAutomaticVersioning(true))
 		switch {
 		case err != nil:
-			diags = append(diags, diag.Diagnostic{Severity: diag.Error, Summary: "error setting principals", Detail: err.Error()})
+			return diag.Errorf("error setting principals on role: %v", err)
```

Comment:
Sorry for reviewing this bit by bit like this, but this is a change in behavior that's kind of a worse experience. I think in this function we should probably keep the old behavior of showing all errors rather than just the first one.

Reply:
It's all good, it's just apart of the process to me. Thanks for catching these issues :)